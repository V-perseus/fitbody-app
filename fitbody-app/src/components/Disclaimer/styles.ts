import { StyleSheet, Dimensions } from 'react-native'
import globals from '../../config/globals'
import { ms } from 'react-native-size-matters/extend'

const styles = StyleSheet.create({
  container: {
    ...globals.styles.container,
  },
  modalContainer: {
    flex: 1,
    borderWidth: 4,
    // backgroundColor: 'rgba(0, 0, 0, 0.5)',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalContents: {
    alignItems: 'center',
    borderWidth: 1,
    width: Dimensions.get('window').width - 32,
    maxHeight: Dimensions.get('window').height - 32,
    marginHorizontal: 16,
    borderRadius: 8,
    backgroundColor: globals.styles.colors.colorWhite,
    paddingTop: 22,
    paddingBottom: 22,
    paddingHorizontal: 32,
  },
  modalTitle: {
    fontFamily: globals.fonts.secondary.style.fontFamily,
    fontSize: ms(30),
    textAlign: 'center',
    color: globals.styles.colors.colorPink,
    marginTop: 10,
  },
  scrollView: {
    marginBottom: 11,
  },
  bodyText: {
    flex: 1,
  },
  linkText: {
    color: globals.styles.colors.colorPink,
    textDecorationLine: 'none',
  },
  buttonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: 16,
  },
  activeButton: {
    flex: 1,
    height: 48,
    backgroundColor: globals.styles.colors.colorPink,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    marginRight: 5,
  },
  activeLabel: {
    fontSize: 14,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    color: globals.styles.colors.colorWhite,
  },
  inactiveButton: {
    flex: 1,
    height: 48,
    borderColor: globals.styles.colors.colorBlack,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    marginLeft: 5,
  },
  inactiveLabel: {
    fontSize: 14,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    color: globals.styles.colors.colorBlack,
  },
})

export default styles
