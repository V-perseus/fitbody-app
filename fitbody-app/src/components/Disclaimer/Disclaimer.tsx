import React from 'react'
import { Linking, Pressable, ScrollView, Text, View } from 'react-native'
import * as Animatable from 'react-native-animatable'
import { ms } from 'react-native-size-matters/extend'
import HTML from 'react-native-render-html'

import Health from '../../../assets/images/svg/icon/40px/health.svg'
import globals from '../../config/globals'
import styles from './styles'
import { DisclaimerOptionsProps } from '../../config/routes/routeTypes'

interface IDisclaimerProps extends DisclaimerOptionsProps {}
const Disclaimer: React.FC<IDisclaimerProps> = ({ route, navigation }) => {
  const { title, body, acceptHandler, approvalRequired } = route.params

  function onAccept() {
    acceptHandler()
  }

  function declineHandler() {
    navigation.goBack()
  }

  return (
    <View style={styles.container}>
      <View style={styles.modalContainer}>
        <Animatable.View animation="fadeInUp">
          <View style={styles.modalContents}>
            <Health />
            <Text style={styles.modalTitle}>{title}</Text>
            <ScrollView style={styles.scrollView}>
              <HTML
                html={body}
                onLinkPress={(_, href) => Linking.openURL(href)}
                listsPrefixesRenderers={{
                  ul: () => {
                    return (
                      <Text
                        style={{ backgroundColor: globals.styles.colors.colorPink, width: 6, height: 6, marginTop: 6, marginRight: 8 }}
                      />
                    )
                  },
                }}
                tagsStyles={{
                  p: {
                    fontSize: ms(14),
                    marginVertical: 11,
                    textAlign: 'center',
                  },
                  a: {
                    textDecorationLine: 'none',
                  },
                  ul: {
                    paddingLeft: 0,
                    fontSize: ms(12),
                    marginBottom: 0,
                  },
                }}
              />
            </ScrollView>
            {approvalRequired ? (
              <View style={styles.buttonContainer}>
                <Pressable style={styles.activeButton} onPress={onAccept}>
                  <Text style={styles.activeLabel}>ACCEPT</Text>
                </Pressable>
                <Pressable style={styles.inactiveButton} onPress={declineHandler}>
                  <Text style={styles.inactiveLabel}>DECLINE</Text>
                </Pressable>
              </View>
            ) : (
              <View style={styles.buttonContainer}>
                <Pressable style={styles.activeButton} onPress={onAccept}>
                  <Text style={styles.activeLabel}>GOT IT!</Text>
                </Pressable>
              </View>
            )}
          </View>
        </Animatable.View>
      </View>
    </View>
  )
}

export default Disclaimer
