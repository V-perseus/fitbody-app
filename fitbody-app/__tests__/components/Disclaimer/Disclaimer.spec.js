import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import Disclaimer from '../../../src/components/Disclaimer'

describe('Disclaimer', () => {
  const mockNavigate = jest.fn()
  const onAccept = jest.fn()

  afterEach(() => {
    cleanup()
  })

  it('renders snapshot correctly', () => {
    const tree = renderer
      .create(
        <Disclaimer
          route={{
            params: {
              title: 'TITLE',
              body: '<div>BODY</div>',
              acceptHandler: onAccept,
              approvalRequired: false,
            },
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders content and buttons', () => {
    const { getByText } = render(
      <Disclaimer
        navigation={{ goBack: mockNavigate }}
        route={{
          params: {
            title: 'TITLE',
            body: '<div>BODY</div>',
            acceptHandler: onAccept,
            approvalRequired: true,
          },
        }}
      />,
    )

    const titleText = getByText('TITLE')
    const bodyText = getByText('BODY')
    expect(titleText).not.toBeNull()
    expect(bodyText).not.toBeNull()

    const acceptBtn = getByText('ACCEPT')
    const declineBtn = getByText('DECLINE')
    expect(acceptBtn).not.toBeNull()
    expect(declineBtn).not.toBeNull()

    fireEvent.press(acceptBtn)
    expect(onAccept).toBeCalledTimes(1)

    fireEvent.press(declineBtn)
    expect(mockNavigate).toBeCalledTimes(1)
  })

  it('renders when approval not required', () => {
    const { getByText } = render(
      <Disclaimer
        route={{
          params: {
            title: 'TITLE',
            body: '<div>BODY</div>',
            acceptHandler: onAccept,
            approvalRequired: false,
          },
        }}
      />,
    )
    const gotItBtn = getByText('GOT IT!')
    expect(gotItBtn).not.toBeNull()
    fireEvent.press(gotItBtn)
    expect(onAccept).toBeCalledTimes(2)
  })
})
