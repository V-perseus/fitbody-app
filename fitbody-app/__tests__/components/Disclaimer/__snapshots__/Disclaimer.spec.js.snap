// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer renders snapshot correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderWidth": 4,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      forwardedRef={[Function]}
      style={
        {
          "opacity": 0,
          "transform": [
            {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 16,
            "maxHeight": 1302,
            "paddingBottom": 22,
            "paddingHorizontal": 32,
            "paddingTop": 22,
            "width": 718,
          }
        }
      >
        <health.svg />
        <Text
          style={
            {
              "color": "#fa8dd5",
              "fontFamily": "BebasNeue",
              "fontSize": 42.173913043478265,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          TITLE
        </Text>
        <RCTScrollView
          style={
            {
              "marginBottom": 11,
            }
          }
        >
          <View>
            <View
              style={{}}
            >
              <View
                style={
                  [
                    {},
                    {},
                    {},
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  selectable={false}
                  style={
                    [
                      {},
                      {},
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    BODY
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 16,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fa8dd5",
                "borderRadius": 8,
                "flex": 1,
                "height": 48,
                "justifyContent": "center",
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffffff",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 14,
                }
              }
            >
              GOT IT!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
